src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java

```diff
@@ -156,11 +161,13 @@ public void check(GHPullRequest ghpr) {
         checkSkipBuild(pr);
```

Comment:
I think this (and tryBuild) needs to be under a synch too, because shouldRun will be updated.